tools/shell/embedded_shell.cpp

```diff
@@ -774,6 +774,9 @@ std::string EmbeddedShell::printJsonExecutionResult(QueryResult& queryResult) co
         printString = jsonDrawingCharacters->ArrayOpen;
     }
     while (queryResult.hasNext()) {
+        if (printInterrupted) {
```

Comment:
I wonder if this printInterrupted check should be outside the while loop. Also we seem to add this check in multiple print function (e.g. latex, html, ...). I wonder if there should be a refactor so that we can check once and then trigger different type of printing.

Reply:
Yeah, I was thinking about that too. The reason why it is inside the while loop and in so many functions is because if the user sets their printing to print a higher number of tuples (for example they want more rows printing for csv output), if we don't check inside the while loop, they can't terminate the printing by pressing ctrl c. If you have a better idea for how to combat this let me know 👍

Reply:
I see. I don't have a good idea here either. In such case you can go ahead and merge.